python/cuml/cluster/__init__.py

```diff
@@ -19,4 +19,5 @@
 from cuml.cluster.agglomerative import AgglomerativeClustering
 from cuml.cluster.hdbscan import HDBSCAN
 from cuml.cluster.hdbscan import condense_hierarchy
-from cuml.cluster.hdbscan import all_points_membership_vectors
+from cuml.cluster.prediction import all_points_membership_vectors
+from cuml.cluster.prediction import approximate_predict
```

Comment:
Similar to https://github.com/rapidsai/cuml/pull/4872#discussion_r956470116 , I see we already have `condense_hierarchy` here but I think we want these exposed not in the `cuml.cluster.*` namespace but in the `cuml.cluster.hdbscan.*` namespace, for two reasons.
- This functionality is tightly coupled to HDBSCAN. Being in the `cluster` namespace implicitly suggests it applies more broadly
- Today, switching between the CPU hdbscan and cuML hdbscan backend is necessary for developers who want to provide both functionality. Having to use separate backend namespaces for the clusterer and the functionality is annoying. As an example of what's necessary in this setup, see [this gist](https://gist.github.com/beckernick/330fbed460061c7f64cfa9929111cdd1#file-benchmark-membership-vectors-py-L54-L72). I had to use two different if/else branches to choose a backend depending on what I was doing.